### PR TITLE
IWYU update

### PR DESCRIPTION
**Pull request by EvoPulseGaming**
Fixed an issue not present until you tried to add new headers/classes
Found and removed an additional monolithic header

### CODE REVIEW

File: Plugins/OceanPlugin/Source/OceanPlugin/OceanPlugin.Build.cs [hunk 1]
```diff
@@ -38,12 +38,8 @@ public OceanPlugin(ReadOnlyTargetRules Target) : base(Target)
         PrivateIncludePaths.AddRange(new string[] { "OceanPlugin/Private" });
         // PublicIncludePaths.AddRange(new string[] { "OceanPlugin/Public" });
 
-        //Enable IWYU but keep our PrivatePCH in use
```
Comment (TheZoc) [inline Plugins/OceanPlugin/Source/OceanPlugin/OceanPlugin.Build.cs:41]: Why remove these?

**Comment by TheZoc** [comment]
Oh, I missed this pull request.

Why remove `PCHUsageMode.UseExplicitOrSharedPCHs` ?

Apparently, you need it to enforce IWYU.

https://docs.unrealengine.com/latest/INT/Programming/UnrealBuildSystem/IWYUReferenceGuide/index.html

**Comment by EvoPulseGaming** [comment]
Nevermind, clearly I made an error, if you want though, you could remove the 1 header. I'll undo my commit on my repo

**Comment by TheZoc** [comment]
Thank you very much, again, @saschaelble!

I added those changes and more in 81566cd65d10c9dfbee827f35f6f9f529482a661 !

After reading [UE4's IWYU Reference Guide](https://docs.unrealengine.com/en-us/Programming/UnrealBuildSystem/IWYUReferenceGuide), I noticed that a non-unity build with PCH disabled was necessary to check if all the includes were correct and I fixed all the issues that appeared. I think the project is finally fully IWYU compliant 😄

**Pull request closed without merging**